pkg/bootstrapper/service.go

```diff
@@ -0,0 +1,258 @@
+	}
+	return nil
+}
+
+// startDependentService starts the kubelet dependent Windows service
+func startDependentService(serviceObj *mgr.Service) error {
+	if serviceObj == nil {
+		return fmt.Errorf("no kubelet dependent service found")
+	}
+	isServiceRunning, err := isDependentRunning(serviceObj)
+	if err != nil {
+		return fmt.Errorf("unable to check if service is running: %v", err)
+	}
+	if !isServiceRunning {
+		err := serviceObj.Start()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// controlService sends a signal to the service and waits until it changes state in response to the signal
+func (service *kubeletService) controlService(cmd svc.Cmd, desiredState svc.State) error {
+	status, err := service.obj.Control(cmd)
+	if err != nil {
+		return err
+	}
+	// Most of the rest of the function borrowed from the package (golang.org/x/sys/windows/svc/mgr) example
+	// Arbitrary wait time
+	timeout := time.Now().Add(serviceWaitTime)
+	for status.State != desiredState {
+		if timeout.Before(time.Now()) {
+			return fmt.Errorf("timeout waiting for service to go to state=%d", desiredState)
+		}
+		time.Sleep(300 * time.Millisecond)
+		status, err = service.obj.Query()
+		if err != nil {
+			return fmt.Errorf("could not retrieve service status: %v", err)
+		}
+	}
+	return nil
+}
+
+// controlDependentService sends a control signal to the kubelet dependent service
+func controlDependentService(serviceObj *mgr.Service, cmd svc.Cmd, desiredState svc.State) error {
+	status, err := serviceObj.Control(cmd)
+	if err != nil {
+		return err
+	}
+	// Most of the rest of the function borrowed from the package (golang.org/x/sys/windows/svc/mgr) example
+	// Arbitrary wait time
+	timeout := time.Now().Add(serviceWaitTime)
+	for status.State != desiredState {
+		if timeout.Before(time.Now()) {
+			return fmt.Errorf("timeout waiting for service to go to state=%d", desiredState)
+		}
+		time.Sleep(300 * time.Millisecond)
+		status, err = serviceObj.Query()
+		if err != nil {
+			return fmt.Errorf("could not retrieve service status: %v", err)
+		}
+	}
+	return nil
+}
+
+// stopKubeletService stops the Windows service via the Windows service API
+func (service *kubeletService) stopKubeletService() error {
+	isServiceRunning, err := service.isKubeletServiceRunning()
+	if err != nil {
+		return fmt.Errorf("unable to check if kubelet service is running: %v", err)
+	}
+	if !isServiceRunning {
+		return nil
+	}
+
+	if len(service.svcDependents) != 0 {
+		for _, dependent := range service.svcDependents {
+			if err := stopDependentService(dependent); err != nil {
+				return fmt.Errorf("failed to stop services dependent on %s", KubeletServiceName)
+			}
+		}
+	}
+
+	if err := service.controlService(svc.Stop, svc.Stopped); err != nil {
+		return fmt.Errorf("unable to stop Windows Service %s", KubeletServiceName)
+	}
+
+	return nil
+}
+
+// stopDependentService stops the kubelet dependent service
+func stopDependentService(serviceObj *mgr.Service) error {
```

Comment:
Mentions kubelet service but this seems generic.

in addition should this be renamed to `stopDependantServices`? As there could be more than one dependant service